source/Glimpse.Ado/AlternateType/GlimpseDbProviderFactory.cs

```diff
@@ -31,9 +33,27 @@ public override bool CanCreateDataSourceEnumerator
 
         private TProviderFactory InnerFactory { get; set; }
 
+        private static bool ShouldNotWrap()
+        {
```

Comment:
Ok, I added this functionality and added the call. However I left the first check against the MessageBroker, since I know that is going to be null when Glimpse HttpHandler is not configured at all in the Web.config file (not sure what the default runtime policy would be in that case?). Mostly to make sure in production cases where it is off, it short circuits as soon as possible with the least amount of overhead.

If the runtime policy is going to be OFF if not configured in the web.config file as handler, then maybe the MessageBroker check is not needed?